File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def helper(self, token):
        try:
            int(token)
            return True
        except:
            return False

    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            if self.helper(token):
                stack.append(int(token))
            else:
                n2 = stack.pop()
                n1 = stack.pop()
                if token == "+":
                    stack.append(n1+n2)
                elif token == "-":
                    stack.append(n1-n2)
                elif token == "*":
                    stack.append(n1*n2)
                elif token == "/":
                    stack.append(int(n1/n2))
        
        return stack[0]

File: Data Structures & Algorithms/validate-parentheses/submission-2.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        Map<Character, Character> cache = new HashMap<>(Map.of(
            ')','(',
            ']','[',
            '}','{'
            ));
        
        for (char c : s.toCharArray()){
            if(cache.containsKey(c)){
                if(!stack.isEmpty() && stack.peek() == cache.get(c)){
                    stack.pop();
                } else {
                    return false;
                }
            } else {
                stack.push(c);
            }
        }
        return stack.isEmpty();

    }
}
